charts/argo-cd/templates/argocd-server/deployment.yaml

```
@@ -33,55 +33,209 @@ spec:
       imagePullSecrets:
         {{- toYaml . | nindent 8 }}
       {{- end }}
-      {{- if .Values.global.securityContext }}
-      securityContext: {{- toYaml .Values.global.securityContext | nindent 8 }}
+      {{- with .Values.global.securityContext }}
+      securityContext: {{- toYaml . | nindent 8 }}
```

Not a blocker but the container-level security context is defined like this
```suggestion
      securityContext:
        {{- toYaml . | nindent 8 }}
```

---

Done